### PR TITLE
Compatible with async-std.

### PR DESCRIPTION
**Pull request by gmg137**
- Compatible with async-std.
- Fix test errors.

Note: reqwest is pure rust, isahc relies on curl.

### CODE REVIEW

**Comment by Empty2k12** [comment]
Hello! Thanks for openig this PR! I think it makes a lot of sense to support all runtimes available.

DO you think, however, we could support `async-std` only with a feature-flag and use `reqwest` when users do not opt into `async-std`. I think we can select the http client dynamically via the feature flag. Tests in CI would nbeed to run twice, once with each http library.

**Comment by gmg137** [comment]
hello, providing multiple optional http clients is a good idea, but it can add to the complexity of use. The user does not perceive the difference between reqwest and isahc in terms of functionality, so whether or not to use feature flag requires you to make the trade-off.

**Comment by Empty2k12** [comment]
`reqwest` is pure rust, while `isahc` is dependent on the OS. I think that should be opt-in :) Interms of complexity, maybe a common HTTP Abstraction for the library could be provided which takes care of sending requests with `isahc` or `reqwest` depending on the feature-flag.

**Comment by gmg137** [comment]
Indeed, it makes more sense to offer an optional feature flag. This requires you to redo the global feature design and update the instructions file. You can just change my PR, or tell me what needs to be done.

File: influxdb/src/client/mod.rs [hunk 1]
```diff
@@ -222,24 +223,24 @@ impl Client {
                 url.query_pairs_mut()
                     .append_pair("precision", &write_query.get_precision());
 
-                ReqwestClient::new().post(url).body(query.get())
+                isahc::post_async(query.get(), url.as_str().to_owned()).await
```
Comment (MTRNord) [inline influxdb/src/client/mod.rs:226]: If my IDE doesn't trick me the parameters are the wrong way around. First argument should be url and second should be body.

File: influxdb/src/client/mod.rs [hunk 2]
```diff
@@ -205,9 +206,9 @@ impl Client {
                 url.query_pairs_mut().append_pair("q", &read_query);
 
                 if read_query.contains("SELECT") || read_query.contains("SHOW") {
-                    ReqwestClient::new().get(url)
+                    isahc::get_async(url.as_str()).await
                 } else {
-                    ReqwestClient::new().post(url)
+                    isahc::post_async("", url.as_str().to_owned()).await
```
Comment (MTRNord) [inline influxdb/src/client/mod.rs:211]: Same order of arguments here too according to my IDE

**Comment by Empty2k12** [comment]
@gmg137 How do you think should the global feature design be changed? I thought it might be possible to add a feature flag for toggling between async-std/tokio

**Comment by gmg137** [comment]
@Empty2k12 Agreed, adding a feature flag is by far the most common practice.

**Comment by Empty2k12** [comment]
This is now implemented in #72. Thanks for your interest in the project.

**Pull request closed without merging**